src/helpers.coffee

```diff
@@ -115,20 +115,20 @@ exports.keystrokeForKeyboardEvent = (event, dvorakQwertyWorkaroundEnabled) ->
   key = keyForKeyboardEvent(event, dvorakQwertyWorkaroundEnabled)
 
   keystroke = ''
-  if event.ctrlKey
+  if event.ctrlKey or key is 'Control'
     keystroke += 'ctrl'
-  if event.altKey
+  if event.altKey or key is 'Alt'
     keystroke += '-' if keystroke
     keystroke += 'alt'
-  if event.shiftKey
+  if event.shiftKey or key is 'Shift'
     # Don't push 'shift' when modifying symbolic characters like '{'
     unless /^[^A-Za-z]$/.test(key)
       keystroke += '-' if keystroke
       keystroke += 'shift'
-  if event.metaKey
+  if event.metaKey or key is 'Meta'
     keystroke += '-' if keystroke
     keystroke += 'cmd'
-  if key?
+  if key? and not KeyboardEventModifiers.has(key)
```

Comment:
This whole hunk is to deal with the fact that `keyForKeyboardEvent` is no longer returning null for the modifier keys. No duplicate modifiers!